notebooks/237-typo-detector/237-typo-detector.ipynb

```diff
@@ -0,0 +1,537 @@
+{
+ "cells": [
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Typo Detector with OpenVINO™\n",
+    "\n",
+    "Typo detection in AI is a process of identifying and correcting typographical errors in text data using machine learning algorithms. The goal of typo detection is to improve the accuracy, readability, and usability of text by identifying and indicating mistakes made during the writing process. To detect typos, AI-based typo detectors use various techniques, such as natural language processing (NLP), machine learning (ML), and deep learning (DL).\n",
+    "\n",
+    "A typo detector takes a sentence as an input and identify all typographical errors such as misspellings and homophone errors.\n",
+    "\n",
+    "This tutorial provides how to use the [Typo Detector](https://huggingface.co/m3hrdadfi/typo-detector-distilbert-en) from the [Hugging Face Transformers](https://huggingface.co/docs/transformers/index) library in the OpenVINO environment to perform the above task.\n",
+    "\n",
+    "The model detects typos in a given text with a high accuracy having these evaluation performances,\n",
+    "- Precision score of 0.9923\n",
+    "- Recall score of 0.9859\n",
+    "- f1-score of 0.9891\n",
+    "\n",
+    "These metrics indicate that the model can correctly identify a high proportion of both correct and incorrect text, minimizing both false positives and false negatives.\n",
+    "\n",
+    "The model has been pretrained on the [NeuSpell](https://github.com/neuspell/neuspell) dataset.\n",
+    "\n",
+    "<br />\n",
+    "\n",
+    "<img src=https://user-images.githubusercontent.com/80534358/224564463-ee686386-f846-4b2b-91af-7163586014b7.png>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Imports"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from transformers import AutoConfig, AutoTokenizer, AutoModelForTokenClassification, pipeline\n",
+    "from pathlib import Path\n",
+    "import numpy as np\n",
+    "import torch\n",
+    "import re"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Methods\n",
+    "\n",
+    "There are two methods to use the typo detection model with OpenVino. In this tutorial we will look at both of them.\n",
+    "\n",
+    "##### 1. Using the [Hugging Face Optimum](https://huggingface.co/docs/optimum/index) library\n",
+    "The Hugging Face Optimum API is a high-level API that allows us to convert and quantize models from the Hugging Face Transformers library to the OpenVINO™ IR format. Optimum allows to the use of optimization on targeted hardware.\n",
```

Comment:
"The Hugging Face Optimum API is a high-level API that allows us to convert and quantize models from the Hugging Face Transformers library to the OpenVINO™ IR format." What does this mean? Does it mean that when you import this package, like "OVModelForTokenClassification", Optimum has already converted and quantized the model for you? So it's a model quantized into int8 model?

Also, "Optimum allows to the use of optimization on targeted hardware", for this point, OpenVINO IR files also targeted on doing this. If it's mentioned here, then it should also be mentioned on the 2nd method.

Reply:
I'm sorry for the confusion. Quantization is actually not used here. I removed that part.
I mentioned the hardware optimization in the 2nd method as well.